mne/preprocessing/maxwell.py

```diff
@@ -771,31 +761,32 @@ def _update_sensor_geometry(info, cal_chans):
     """Helper to replace sensor geometry information"""
 
     # Replace sensor locations (and track differences) for fine calibration
-    ang_shift = np.zeros((info['nchan'], 3))
-    for cal_chan in cal_chans:
-        ch_ind = info['ch_names'].index(cal_chan['ch_name'])
-
-        # Loop over vectors in X, Y, and Z directions
-        for ang_i, (i1, i2) in enumerate(zip(range(3, 10, 3),
-                                             range(6, 13, 3))):
-            v1 = cal_chan['loc'][i1:i2]
-            v2 = info['chs'][ch_ind]['loc'][i1:i2].astype(float)
-
-            cos_ang = np.dot(v1, v2) / (np.linalg.norm(v1) *
-                                        np.linalg.norm(v2))
-            ang_shift[ch_ind, ang_i] = cos_ang
+    ang_shift = np.zeros((len(cal_chans), 3))
+    used = np.zeros(len(info['chs']))
+    for ci, cal_chan in enumerate(cal_chans):
+        idx = info['ch_names'].index(cal_chan['ch_name'])
+        assert not used[idx]
+        used[idx] = True
+        info_chan = info['chs'][idx]
+
+        # calculate shift angle
+        v1 = cal_chan['coil_trans'][:3, :3]
+        _normalize_vectors(v1)
+        v2 = info_chan['coil_trans'][:3, :3]
+        _normalize_vectors(v2)
+        ang_shift[ci] = np.sum(v1 * v2, axis=1)
 
         # Adjust channel orientation with those from fine calibration
-        info['chs'][ch_ind]['loc'] = cal_chan['loc']
-        info['chs'][ch_ind]['coil_trans'] = cal_chan['coil_trans']
+        info_chan['coil_trans'] = cal_chan['coil_trans']
+        assert info_chan['coord_frame'] == cal_chan['coord_frame']
 
```

Comment:
Actually I don't think the current version actually replaced the `loc` anyway. Hmm...